components/browser/session/src/main/java/mozilla/components/browser/session/SessionManager.kt

```diff
@@ -20,6 +22,7 @@ class SessionManager(
     delegate: Observable<SessionManager.Observer> = ObserverRegistry()
 ) : Observable<SessionManager.Observer> by delegate {
     private val values = mutableListOf<Session>()
+    // Access must be guarded by synchronizing on `values`.
```

Comment:
There's `@GuardedBy` annotation that may be helpful:
https://developer.android.com/reference/android/support/annotation/GuardedBy